api/channel.go

```diff
@@ -824,8 +824,24 @@ func updateLastViewedAt(c *Context, w http.ResponseWriter, r *http.Request) {
 	params := mux.Vars(r)
 	id := params["channel_id"]
 
+	doClearPush := false
+	if *utils.Cfg.EmailSettings.SendPushNotifications && !c.Session.IsMobileApp() {
```

Comment:
maybe I'm missing something, but the badge icon counts is for all the channels.  Wouldn't you need to loop through all the channels?

This looks like you're clearing based on one channel?